tests/components/srp_energy/test_config_flow.py

```diff
@@ -1,120 +1,121 @@
 """Test the SRP Energy config flow."""
-from unittest.mock import patch
+from unittest.mock import MagicMock
 
-from homeassistant import config_entries, data_entry_flow
-from homeassistant.components.srp_energy.const import CONF_IS_TOU, SRP_ENERGY_DOMAIN
+from homeassistant import config_entries
+from homeassistant.components.srp_energy import CONF_IS_TOU, DOMAIN
+from homeassistant.const import CONF_ID, CONF_PASSWORD, CONF_SOURCE, CONF_USERNAME
 from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
 
-from . import ENTRY_CONFIG, init_integration
+from . import ACCNT_ID, ACCNT_IS_TOU, ACCNT_PASSWORD, ACCNT_USERNAME, TEST_USER_INPUT
 
+from tests.common import MockConfigEntry
 
-async def test_form(hass: HomeAssistant) -> None:
-    """Test user config."""
-    # First get the form
+
+async def test_show_form(hass, mock_srp_energy_config_flow: MagicMock, capsys) -> None:
+    """Test show configuration form."""
     result = await hass.config_entries.flow.async_init(
-        SRP_ENERGY_DOMAIN, context={"source": config_entries.SOURCE_USER}
+        DOMAIN, context={CONF_SOURCE: config_entries.SOURCE_USER}
     )
-    assert result["type"] == data_entry_flow.FlowResultType.FORM
+
+    assert result["type"] == FlowResultType.FORM
     assert result["step_id"] == "user"
     assert result["errors"] == {}
 
-    # Fill submit form data for config entry
-    with patch(
-        "homeassistant.components.srp_energy.config_flow.SrpEnergyClient"
-    ), patch(
-        "homeassistant.components.srp_energy.async_setup_entry",
-        return_value=True,
-    ) as mock_setup_entry:
-        result = await hass.config_entries.flow.async_configure(
-            result["flow_id"],
-            user_input=ENTRY_CONFIG,
-        )
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
We still want to patch the integration setup function to avoid setting up the whole integration.